test/s3select_test.cpp

```diff
@@ -629,3 +630,130 @@ TEST(TestS3selectFunctions, modzero)
+} 
+
+TEST(TestS3selectFunctions, nullifnull)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select nullif(null,null) from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string(",\n"));
+} 
+
+TEST(TestS3selectFunctions, nullifstring)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select nullif(\"james\",\"bond\") from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("james,\n"));
+} 
+
+TEST(TestS3selectFunctions, nullifeqstring)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select nullif(\"redhat\",\"redhat\") from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string(",\n"));
+} 
+
+TEST(TestS3selectFunctions, nulladdition)
+{
+    s3select s3select_syntax;
+    const std::string input_query = "select 1 + null from stdin;";
+    auto status = s3select_syntax.parse_query(input_query.c_str());
+    ASSERT_EQ(status, 0);
+    s3selectEngine::csv_object s3_csv_object(&s3select_syntax);
+    std::string s3select_result;
+    std::string input;
+    size_t size = 1;
+    generate_csv(input, size);
+    status = s3_csv_object.run_s3select_on_object(s3select_result, input.c_str(), input.size(), 
+        false, // dont skip first line 
+        false, // dont skip last line
+        true   // aggregate call
+        ); 
+    ASSERT_EQ(status, 0); 
+    ASSERT_EQ(s3select_result, std::string("nan,\n"));
```

Comment:
is the result of arith + null defined to be nan in SQL standards?  I thought the result of such an operation was null.

Reply:
[here](https://docs.google.com/document/d/1edRKxCcbxqNNyH-CsHt3s870eiBdMMjFZknbSVhkOOo/edit?usp=sharing) , it was note that arithmetic operation with NULL return NAN
the thought behind that, is the fact that NAN is supported in C/C++ as for operations (arithmetic and compare) with NAN

however, we could define other rule, such as return NULL for those operations , which mean unknown.
currently, i'm not sure what is the sql-standard for that.

Reply:
I guess ISO/IEC 9075;  It's of course, not free to download.

Reply:
ok, will review that.
it means that 
1+null -> null 
1+nan -> nan (built-in c/c++) 

1 + null + nan -> null (?) 
nan + 1 + null -> nan (?)

need to decide which (null/nan) got precedence , and stick to that.

Reply:
I suspect:  any expression involving nan, including 0 or more null terms, results in NaN